File: app/src/main/java/com/work/criminalintent/Crime.java
package com.work.criminalintent;

import java.util.Date;
import java.util.UUID;
import lombok.Data;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@ToString
public class Crime {
    @Getter
    private UUID id;
    @Getter
    @Setter
    private String title;
    @Getter
    @Setter
    private Date date;
    @Getter
    @Setter
    private boolean solved;
    @Getter
    @Setter
    private boolean requiresPolice;
    public Crime() {
        id = UUID.randomUUID();
        date = new Date();
    }
}

File: app/src/main/java/com/work/criminalintent/kt/Crime.kt
package com.work.criminalintent.kt

import java.util.*

class Crime @JvmOverloads constructor(val id: UUID = UUID.randomUUID()) {
    var title: String? = null
    var date = Date()
    var isSolved = false
    var suspect: String? = null
    fun getPhotoFileName(): String = "IMG_$id.jpg"
}
